### PR TITLE
Added huject types

### PR DESCRIPTION
**Pull request by alfaslash**
**Added huject types**.

### CODE REVIEW

**Comment by vvakame** [comment]
could you rename `huject/huject-test.ts` to `huject/huject-tests.ts`?

File: huject/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "target": "es6",
+        "noImplicitAny": true,
+        "strictNullChecks": false,
```
Comment (vvakame) [inline huject/tsconfig.json]: try using `"strictNullChecks": true,`

File: huject/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+     * @Factory
+     * class MyModelFactory {
+     *     @Factory
+     *     public createModel(): Model {return null;}
+     * }
+     */
+    export function Factory<T>(target: Object, propertyKey: string|symbol, descriptor: TypedPropertyDescriptor<T>): TypedPropertyDescriptor<T> | void;
+}
+
+declare module "huject" {
```
Comment (vvakame) [inline huject/index.d.ts]: could you strip `declare module "huject" {`?
https://github.com/Microsoft/types-publisher#fields-in-datadefinitionsjson
> ProperModule: Only top-level import and export declarations. Preferred

File: huject/tsconfig.json (hunk 1 above)
Comment (alfaslash) [inline huject/tsconfig.json]: Done. Thanks.

File: huject/index.d.ts (hunk 2 above)
Comment (alfaslash) [inline huject/index.d.ts]: Done.

**Comment by vvakame** [comment]
ah, sorry.
https://github.com/asvetliakov/huject
huject package have [own bundle](https://github.com/asvetliakov/Huject/blob/master/huject.d.ts)
We recommend you to send this change to there repository.
http://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

**Comment by minestarks** [comment]
@alfaslash Thank you for your contribution. I'm closing out this PR as it's been inactive for a couple of weeks and the changes seem to belong to the official typings maintained by the project.

**Pull request closed without merging**